tensorboard/tools/mat_bundle_icon_svg.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# Exposes input SVGs as definitions with filename, without extension, as the
+# ids.
+# Usage: mat_bundle_icon_svg.sh [output.svg] [in_1.svg] [in_2.svg] ...
+
+OUTPUT_PATH=$1
+shift
+
+PREAMBLE='<svg><defs>'
+POSTAMBLE='</defs></svg>'
+
+echo $PREAMBLE > $OUTPUT_PATH
+for file in "$@"; do
+  svg_id="$(basename $file | sed 's|\.svg$||g')"
+  cat $file | sed "s|<svg|<svg id=\"$svg_id\"|g" >> $OUTPUT_PATH
```

Comment:
You can avoid re-opening the file *n* times by putting the whole thing
inside a redirection:

```shell
{
    echo "${PREAMBLE}"
    for file; do
        ...
    done
    echo "${POSTAMBLE}"
} >"${OUTPUT_PATH}"
```

Reply:
I don't feel too good about the `sed ... 2>&0` part. Is it correct?

Reply:
The code looks good without `2>&0`, but I, too, find that a bit
confusing. The effect of `2>&0` is to let you _read_ from stderr as if
it were stdin; i.e., it’s exactly equivalent to `2<&0`. This isn’t
usually useful, because most programs don’t try to read from stderr.

It should work without any redirections: `sed` will read from `"$file"`
and print to stdout, and stdout is redirected to `"${OUTPUT_PATH}"` in
the outer `{ … }` group.